frontend/src/pages/modelServing/screens/types.ts

```diff
@@ -91,6 +93,7 @@ export type InferenceServiceStorage = {
   type: InferenceServiceStorageType;
   path: string;
   dataConnection: string;
+  connectionType?: ModelLocationType | string;
```

Comment:
As discussed on the call with @alexcreasy - If possible, we probably want to avoid adding model-registry-specific properties to these model serving types. `isRecommended` / `LabeledConnection` is sort of an exception because it's a generic feature we added to model serving - anything can make a connection recommended.

I know this complicates things though - I want to think through it a little but we could decide to just address it outside this PR. If we do, let's put a comment here and anywhere this is used saying it's temporary.

Reply:
For posterity: tech debt-issue: https://issues.redhat.com/browse/RHOAIENG-19991(also added in the comments)